docs: Add new section "Running a Node" (#7701)
cosmos/cosmos-sdk: docs/run-node/run-node.md

<!--
order: 2
-->

# Running a Node

> Now that the application is ready and the keyring populated, it's time to see how to run the blockchain node. In this section, the application we are running is called [`simapp`](https://github.com/cosmos/cosmos-sdk/tree/v0.40.0-rc1/simapp), and its corresponding CLI binary `simd`. {synopsis}

Reviewer: ```suggestion
Now that the application is ready and the keyring populated, it's time to see how to run the blockchain node. In this section, the application we are running is called [`simapp`](https://github.com/cosmos/cosmos-sdk/tree/v0.40.0-rc2/simapp), and its corresponding CLI binary `simd`. {synopsis}
```

:::

Before actually running the node, we need to initialize the chain, and most importantly its genesis file. This is done with the `init` subcommand:

```bash
# The argument <moniker> is the custom username of your node, it should be human-readable.
simd init <moniker> --chain-id my-test-chain
```

The command above creates all the configuration files needed for your node to run, as well as a default genesis file, which defines the initial state of the network. All these configuration files are in `~/.simapp` by default, but you can overwrite the location of this folder by passing the `--home` flag.

The `~/.simapp` folder has the following structure:

```bash
.                                   # ~/.simapp
  |- data                           # Contains the databases used by the node.
  |- config/
      |- app.toml                   # Application-related configuration file.
      |- config.toml                # Tendermint-related configuration file.
      |- genesis.json               # The genesis file.
      |- node_key.json              # Tendermint node pubkey that will be used for signing blocks.

>       |- priv_validator_key.json    # TODO

Author: Does someone have a link to what this file does? In particular, I don't know the difference between this file and the `node_key.json` one.

Reviewer 2: ```
# Path to the JSON file containing the private key to use as a validator in the consensus protocol
priv_validator_key_file = "config/priv_validator_key.json"

# Path to the JSON file containing the private key to use for node authentication in the p2p protocol
node_key_file = "config/node_key.json"
```

```

Before starting the chain, you need to populate the state with at least one account. To do so, first [create a new account in the keyring](./keyring.md#adding-keys-to-the-keyring) named `my_validator` (feel free to choose another name).

Now that you have created a local account, go ahead and grant it some `stake` tokens in your chain's genesis file. Doing so will also make sure your chain is aware of this account's existence:

```bash
simd add-genesis-account $(simd keys show my_validator -a) 100000000stake --chain-id my-test-chain
```

> Note that the tokens in the SDK have the `{amount}{denom}` format.

Reviewer 3: Please explain denom. I was initially confused, because I was usually used in context of value denomination. Example: 1USD denominated in US Cent is 100 USCent. Or 1 BTC is denominated in satoshi, where 1 BTC = 1e9 satoshi.

Reviewer: We should also explain how "stake" just happens to be the denom in simapp, and for your own chain with its own staking denom, that should be used instead if following this guide.

Now that your account has some tokens, you need to add a validator to your chain. Validators are special full-nodes that participate in the consensus process (implemented in the [underlying consensus engine](../intro/sdk-app-architecture.md#tendermint)) in order to add new blocks to the chain. Any account can declare its intention to become a validator operator, but only those with sufficient delegation get to enter the active set (for example, only the top 125 validator candidates with the most delegation get to be validators in the Cosmos Hub). For this guide, you will add your local node (created via the `init` command above) as a validator of your chain. Validators can be declared before a chain is first started via a special transaction included in the genesis file called a `gentx`:

```bash
# Create a gentx.
simd gentx --name my_validator --amount 100000stake --chain-id my-test-chain

# Add the gentx to the genesis file.
simd collect-gentxs --chain-id my-test-chain
```

A `gentx` does three things:

> 1. Makes the `validator` account you created into a validator operator account (i.e. the account that controls the validator).

Reviewer 3: ```suggestion
1. Registers the `validator` account you created as a validator operator account (i.e. the account that controls the validator).
```

2. Self-delegates the provided `amount` of staking tokens.
3. Link the operator account with a Tendermint node pubkey that will be used for signing blocks. If no `--pubkey` flag is provided, it defaults to the local node pubkey created via the `simd init` command above.

For more information on `gentx`, use the following command:

```bash
simd gentx --help
```

> Now that everything is set up, you can finally start your node:

Reviewer 3: How about moving this this paragraph into the section below (Run a localnet)?
